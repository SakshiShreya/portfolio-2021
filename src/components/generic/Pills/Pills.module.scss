@import "../../../utilities/partials/size";
@import "../../../utilities/partials/colorCodes";

.pill {
  padding: ($std-spacing / 4) $std-spacing;
  border-radius: 50px;
  color: $white;
  font-size: 12px;
  display: inline-block;
  margin-right: $std-spacing / 2;
  margin-bottom: $std-spacing / 4;
}

.primary {
  background-color: map-get($map: $orange, $key: normal);
}

.secondary {
  background-color: map-get($map: $green, $key: normal);
}

.gray {
  background-color: map-get($map: $gray, $key: normal);
}
