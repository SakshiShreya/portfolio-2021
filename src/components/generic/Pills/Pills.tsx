import * as React from "react";
import styles from "./Pills.module.scss";

interface PillsProps {
  list: Array<IPill>;
  color?: "primary" | "secondary" | "gray"; // Primary by default
}

export interface IPill {
  display: string;
  id: string;
}

const Pills: React.FunctionComponent<PillsProps> = ({ list, color = "primary" }) => {
  return (
    <ul>
      {list.map(pill => (
        <li key={pill.id} className={`${styles.pill} ${styles[color]}`}>
          {pill.display}
        </li>
      ))}
    </ul>
  );
};

export default Pills;
