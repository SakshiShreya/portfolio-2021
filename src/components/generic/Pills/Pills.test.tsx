import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Pills, { IPill } from "./Pills";

const empty: Array<IPill> = [];
  const single: Array<IPill> = [{ display: "Test1", id: "test1" }];
  const multiple: Array<IPill> = [
    { display: "Test1", id: "test1" },
    { display: "Test2", id: "test2" }
  ];

describe("test Pills length", () => {
  it("should render empty pill block if no pill", () => {
    const elem = renderer.create(<Pills list={empty} />).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it("should render single pill", () => {
    const elem = renderer.create(<Pills list={single} />);
    const snapshot = elem.toJSON();
    expect(snapshot).toMatchSnapshot();

    const instance = elem.root;
    expect(instance.findAllByProps({ className: "pill primary" })).toHaveLength(1);
  });

  it("should render multiple pills", () => {
    const elem = renderer.create(<Pills list={multiple} />);
    const snapshot = elem.toJSON();
    expect(snapshot).toMatchSnapshot();

    const instance = elem.root;
    expect(instance.findAllByProps({ className: "pill primary" })).toHaveLength(multiple.length);
  });
});

describe("test Pills color", () => {
  it("should render secondary class if color='secondary' is passed", () => {
    const elem = render(<Pills list={multiple} color="secondary" />).container;
    const children = elem.querySelectorAll("li");

    for(let i = 0; i < children.length; i++) {
      expect(children[i]).toHaveClass("secondary");
    }
  });

  it("should render gray class if color='gray' is passed", () => {
    const elem = render(<Pills list={multiple} color="gray" />).container;
    const children = elem.querySelectorAll("li");

    for(let i = 0; i < children.length; i++) {
      expect(children[i]).toHaveClass("gray");
    }
  });
});
