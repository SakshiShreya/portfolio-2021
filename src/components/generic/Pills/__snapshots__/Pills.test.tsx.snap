// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing of Pills Component should render empty pill block if no pill 1`] = `<ul />`;

exports[`snapshot testing of Pills Component should render multiple pills 1`] = `
<ul>
  <li
    className="pill primary"
  >
    Test1
  </li>
  <li
    className="pill primary"
  >
    Test2
  </li>
</ul>
`;

exports[`snapshot testing of Pills Component should render single pill 1`] = `
<ul>
  <li
    className="pill primary"
  >
    Test1
  </li>
</ul>
`;
